File: backend/src/main/java/com/siheung/seemoney/domain/news/entity/News.java
package com.siheung.seemoney.domain.news.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "news")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class News {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 500)
    private String title;

    @Column(nullable = false, length = 1000)
    private String link;

    @Column(columnDefinition = "TEXT")
    private String summary;

    @Column(length = 100)
    private String pubDate;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: android/app/src/main/java/com/example/siheung_seemoney/data/News.kt
package com.example.siheung_seemoney.data

import com.google.gson.annotations.SerializedName

/**
 * 백엔드에서 내려오는 뉴스 데이터 모델
 */
data class News(
    @SerializedName("title") val title: String,
    @SerializedName("link") val link: String,
    @SerializedName("summary") val summary: String,
    @SerializedName("pubDate") val pubDate: String
)
